internal/gps/source_cache_test.go

```diff
@@ -0,0 +1,426 @@
+// Copyright 2017 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gps
+
+import (
+	"sort"
+	"testing"
+
+	"github.com/golang/dep/internal/gps/pkgtree"
+	"github.com/pkg/errors"
+)
+
+var testAnalyzerInfo = ProjectAnalyzerInfo{
+	Name:    "test-analyzer",
+	Version: 1,
+}
+
+func TestSingleSourceCache(t *testing.T) {
+	const root = "example.com/test"
+
+	t.Run("info", func(t *testing.T) {
+		newSemverConstraint := func(s string) Constraint {
```

Comment:
Didn't you already define the same function in `internal/gps/constraint_test.go`?